icicle/appUtils/poseidon/poseidon.cuh

```diff
@@ -0,0 +1,141 @@
+#pragma once
+#include "../../curves/curve_config.cuh"
+#include "constants.cuh"
+
+#if !defined(__CUDA_ARCH__) && defined(DEBUG)
+#include <iomanip>
+#include <string>
+#include <sstream>
+#include <chrono>
+
+__host__ void print_scalar_t(scalar_t element) {
```

Comment:
In principle, `std::cout` is [overloaded](https://github.com/ingonyama-zk/icicle/blob/main/icicle/primitives/field.cuh#L513) for `scalar_t` already. You can change it to your pretty-print if you want to.

Reply:
Replaced with changing the overload in field.cuh
24222badb6d0db93bc0f0645ef7103beb4a6fa30